buildSrc/src/main/kotlin/org/cqfn/save/buildutils/DockerStackConfiguration.kt

```diff
@@ -90,15 +100,16 @@ fun Project.createStackDeployTask(profile: String) {
             Files.createDirectories(configsDir.resolve("preprocessor"))
         }
         description = "Deploy to docker swarm. If swarm contains more than one node, some registry for built images is required."
-        val args = buildList {
-            add("--compose-file")
-            add("${rootProject.buildDir}/docker-compose.yaml")
+        // this command puts env variables into compose file
+        val composeCmd = "docker-compose -f ${rootProject.buildDir}/docker-compose.yaml --env-file ${rootProject.buildDir}/.env config"
+        val stackCmd = "docker stack deploy --compose-file -" +
             if (useOverride && composeOverride.exists()) {
-                add("--compose-file")
-                add(composeOverride.canonicalPath)
-            }
-        }.toTypedArray()
-        commandLine("docker", "stack", "deploy", *args, "save")
+                " --compose-file ${composeOverride.canonicalPath}"
```

Comment:
in this branch we will have `docker stack deploy --compose-file - --compose-file ${composeOverride.canonicalPath}`?
is this an valid command?

Reply:
No, the complete command will be
```bash
docker-compose --file buiild/docker-compose.yaml --env-file build/.env config | docker stack deploy --compose-file - --compose-file ${composeOverride.canonicalPath}
```
where `--compose-file -` means that config will be read from stdin. `docker-compose` will substitute variables with values from `.env` (apparently, `docker stack` doesn't support this functionality).